### PR TITLE
168736416 controller dbcontext

### PR DESCRIPTION
**Pull request by Smith-Justin**
# 168736416 controller dbcontext

## reviewer

- @fredbelotte

## contributing

this project is maintained under the terms of the [Contribution Guidelines][contribution-guidelines-url].

[contribution-guidelines-url]: https://github.com/revaturexyz/amenityxyz/blob/master/.github/CONTRIBUTING.md 'CONTRIBUTION GUIDELINES'

### CODE REVIEW

File: .docker/dockerup.yaml [hunk 1]
```diff
@@ -15,12 +15,19 @@ services:
     ports:
       - 10080:80
       - 10443:443
+    environment:
```
Comment (fredbelotte) [inline .docker/dockerup.yaml]: use env or app settings not both

File: .docker/dockerup.yaml [hunk 2]
```diff
@@ -15,12 +15,19 @@ services:
...
     deploy:
       replicas: 1
     image: postgres
+    restart: always
     networks:
       - amenityxyz_localhost
+    ports:
+      - 5432:5432
+    environment:
+      PASSWORD: ""
```
Comment (fredbelotte) [inline .docker/dockerup.yaml]: remove this

File: aspnet/Xyz.Amenity.Lib/Models/AmenityModel.cs [hunk 3]
```diff
@@ -1,19 +1,20 @@
...
+using System.ComponentModel.DataAnnotations.Schema;
 using Xyz.Amenity.Lib.Validators;
 
 namespace Xyz.Amenity.Lib.Models
 {
   /// <summary>
   /// The representation of an amenity provided at a location.
   /// </summary>
-  public abstract class AmenityModel : IValidatableObject
+  public class AmenityModel //: IValidatableObject
```
Comment (fredbelotte) [inline aspnet/Xyz.Amenity.Lib/Models/AmenityModel.cs]: remove this commented code

File: aspnet/Xyz.Amenity.Lib/Models/AmenityModel.cs [hunk 4]
```diff
@@ -25,25 +26,22 @@ public AmenityModel()
     /// </summary>
     /// <param name="validationContext"></param>
     /// <returns></returns>
-    public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
+   /*  public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
```
Comment (fredbelotte) [inline aspnet/Xyz.Amenity.Lib/Models/AmenityModel.cs]: remove all these

File: aspnet/Xyz.Amenity.Api/Startup.cs [hunk 5]
```diff
@@ -26,8 +28,10 @@ public Startup(IConfiguration configuration)
         // This method gets called by the runtime. Use this method to add services to the container.
```
Comment (fredbelotte) [inline aspnet/Xyz.Amenity.Api/Startup.cs]: remove this line

File: aspnet/Xyz.Amenity.Api/Startup.cs [hunk 6]
```diff
@@ -26,8 +28,10 @@ public Startup(IConfiguration configuration)
...
         {
-            services.AddMvc().SetCompatibilityVersion(CompatibilityVersion.Version_2_2);
-            services.AddSingleton<IAmenityRepository>(new AmenityRepository());
+
+          services.AddMvc().SetCompatibilityVersion(CompatibilityVersion.Version_2_2);
+          services.AddDbContext<AmenityDbContext>(o => o.UseNpgsql(Configuration.GetConnectionString("DefaultString")));
+          services.AddScoped<IAmenityRepository, AmenityRepository>();
         }
 
         // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
```
Comment (fredbelotte) [inline aspnet/Xyz.Amenity.Api/Startup.cs]: remove this line

File: aspnet/Xyz.Amenity.Api/appsettings.json [hunk 7]
```diff
@@ -4,5 +4,10 @@
       "Default": "Warning"
     }
   },
-  "AllowedHosts": "*"
+  "AllowedHosts": "*",
+  "ConnectionStrings": 
+  {
+    "DefaultString": "host=db;db=postgres;userid=postgres;password="
```
Comment (fredbelotte) [inline aspnet/Xyz.Amenity.Api/appsettings.json]: need a password

File: aspnet/Xyz.Amenity.Dbx/AmenityDbContext.cs [hunk 8]
```diff
@@ -11,6 +11,7 @@ public class AmenityDbContext : DbContext
   {
 
     public DbSet<AmenityEntity> AmenityEntities { get; set; }
+    public DbSet<Restriction> Restrictions {get; set;}
 
     /// <summary>
```
Comment (fredbelotte) [inline aspnet/Xyz.Amenity.Dbx/AmenityDbContext.cs]: what is this single line

Comment (Pronounced) [reply aspnet/Xyz.Amenity.Dbx/AmenityDbContext.cs]: It is needed to create the restrictions table in the DB to store any restrictions that an amenity may have.  Restrictions is a complex object so it needs its own table.

Comment (vitrv) [reply aspnet/Xyz.Amenity.Dbx/AmenityDbContext.cs]: @fredbelotte  would you like us to keep implementing restrictions in this way, or change to something that uses a single table?

File: .docker/dockerup.yaml [hunk 9]
```diff
@@ -20,7 +20,10 @@ services:
     deploy:
       replicas: 1
     image: postgres
+    restart: always
     networks:
       - amenityxyz_localhost
+    environment:
+      PASSWORD: "12345"
```
Comment (fredbelotte) [inline .docker/dockerup.yaml]: what is this?

**Pull request merged**